src/overlays/actors/ovl_En_Nb/z_en_nb.c

```diff
@@ -31,81 +101,659 @@ const ActorInit En_Nb_InitVars = {
+                    this->unk_284 = 0;
+                    break;
+
+                case 0x28CB: // "I told you I already ate!"
+                    EnNb_ChangeAnim(this, EN_NB_ANIM_ANGRY);
+                    break;
+
+                case 0x28C7:
+                case 0x2901:
+                case 0x2902:
+                case 0x2906:
+                case 0x290D:
+                case 0x2912:
+                    EnNb_ChangeAnim(this, EN_NB_ANIM_TALK_LOOP);
+                    break;
+            }
+        }
+
+        this->textId = currentTextId;
+    } else if (this->stateFlags & EN_NB_FLAG_80) {
+        this->unk_18C = NULL;
+        this->textId = 0;
+        this->stateFlags &= ~EN_NB_FLAG_80;
+        EnNb_ChangeAnim(this, EN_NB_ANIM_1);
+    }
+
+    if (this->unk_18C != NULL) {
+        this->unk_18C(this, play);
+    }
+
+    return 0;
+}
+
+s32 func_80BC0B98(EnNb* this, PlayState* play, ScheduleOutput* scheduleOutput) {
+    s32 success = false;
+
+    if (EnNb_FindActor(this, play, ACTORCAT_NPC, ACTOR_EN_AN) != NULL) {
+        SubS_UpdateFlags(&this->stateFlags, EN_NB_FLAG_1 | EN_NB_FLAG_2, EN_NB_FLAG_1 | EN_NB_FLAG_2 | EN_NB_FLAG_4);
+        this->stateFlags |= EN_NB_FLAG_20;
+        EnNb_ChangeAnim(this, EN_NB_ANIM_0);
+        success = true;
+    }
+
+    return success;
+}
+
+s32 func_80BC0C0C(EnNb* this, PlayState* play, ScheduleOutput* scheduleOutput) {
+    if (!(gSaveContext.eventInf[4] & 8)) {
+        SubS_UpdateFlags(&this->stateFlags, EN_NB_FLAG_1 | EN_NB_FLAG_2, EN_NB_FLAG_1 | EN_NB_FLAG_2 | EN_NB_FLAG_4);
+    } else {
+        SubS_UpdateFlags(&this->stateFlags, EN_NB_FLAG_4, EN_NB_FLAG_1 | EN_NB_FLAG_2 | EN_NB_FLAG_4);
+    }
+    EnNb_ChangeAnim(this, EN_NB_ANIM_0);
+
+    return true;
+}
+
+s32 EnNb_ProcessScheduleOutput(EnNb* this, PlayState* play, ScheduleOutput* scheduleOutput) {
+    s32 success;
+
+    this->actor.flags |= ACTOR_FLAG_1;
+    this->actor.targetMode = 0;
+    this->stateFlags = EN_NB_FLAG_NONE;
+    this->unk_274 = 40.0f;
+
+    switch (scheduleOutput->result) {
+        default:
+            success = false;
+            break;
+
+        case EN_NB_SCH_1:
+        case EN_NB_SCH_3:
+        case EN_NB_SCH_4:
+            success = func_80BC0C0C(this, play, scheduleOutput);
+            break;
+
+        case EN_NB_SCH_2:
+            success = func_80BC0B98(this, play, scheduleOutput);
+            break;
+    }
+    return success;
+}
+
+s32 func_80BC0D08(EnNb* this, PlayState* play) {
+    return 0;
+}
+
+void EnNb_HandleSchedule(EnNb* this, PlayState* play) {
+    if ((this->schResultValue == EN_NB_SCH_1) || (this->schResultValue == EN_NB_SCH_2) ||
+        (this->schResultValue == EN_NB_SCH_3) || (this->schResultValue == EN_NB_SCH_4)) {
+        func_80BC0D08(this, play);
+    }
+
+    Math_ApproachS(&this->actor.shape.rot.y, this->actor.world.rot.y, 3, 0x2AA8);
+}
+
+void EnNb_FollowSchedule(EnNb* this, PlayState* play) {
+    ScheduleOutput scheduleResult;
```

Comment:
```suggestion
    ScheduleOutput scheduleOutput;
```
In conjunction with this you could rename `this->schResultValue` to just `this->scheduleResult`.